### PR TITLE
feat(misc): first actual commit

### PR DESCRIPTION
**Pull request by vdice**
Amending:
    `README.md` with actual info

Adding:
    `Dockerfile` for the shell-dev image,
    `Makefile` for the build and push targets,
    `Jenkinsfile` for the minimal CI pipeline

### CODE REVIEW

**Comment by vdice** [comment]
Take a look at the pipeline generated from the `Jenkinsfile` [here](https://ci.deis.io/job/deis%20org/job/docker-shell-dev/branch/add-first-actual-commit/) and the first successful run [here](https://ci.deis.io/job/deis%20org/job/docker-shell-dev/branch/add-first-actual-commit/8/console) (once I got the [pipeline dsl](https://jenkins.io/doc/pipeline/steps/) dialed)

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ withCredentials([[$class: 'StringBinding',
+                    credentialsId: '614c646f-57fc-4cdb-a70e-cbd5b0d91737',
+                    variable: 'QUAY_PASSWORD']]) {
+
+   sh """
+     docker login -e="${quayEmail}" -u="${quayUsername}" -p="\${QUAY_PASSWORD}" quay.io
+     make build push
+   """
+ }
+}
```
Comment (arschles) [inline Jenkinsfile:32]: so. much. 🔥 💥

**Comment by arschles** [comment]
This is awesome @vdice. Is there a description somewhere (doesn't have to be in this repo) that has more info on what the `Jenkinsfile` is, and how to build one?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!groovy
+
+node {
+ stage 'Checkout'
```
Comment (sgoings) [inline Jenkinsfile:4]: "Checkout" isn't maybe the best choice here - maybe "Build"? Is there a standard naming convention for pipeline stages proposed by Jenkins?

Comment (vdice) [inline Jenkinsfile:4]: Note that particular stage just does contain the `checkout scm` step... inspiration from `Jenkinsfile` examples like the one seen [here](https://github.com/jenkinsci/pipeline-examples/blob/master/jenkinsfile-examples/nodejs-build-test-deploy-docker-notify/Jenkinsfile)

That being said, definitely down to adjust wording/organization!

**Comment by vdice** [comment]
Forgot about pushing as well to the `deis` org of the `docker.io` registry; amended PR but need to get actual registry repo created...

File: Jenkinsfile (hunk 2 above)
Comment (sgoings) [inline Jenkinsfile:4]: Let's go with it for now.

**Pull request merged**